imex.go

```diff
@@ -30,6 +30,7 @@ func (i *ImexAgent) DownloadFile() *ImexAgent {
 		i.ErrorData = err
 		return i
 	}
+	defer res.Body.Close()
```

Comment:
Please Check handling res variable please. The res variable can be initialize as nil variable